### ISSUE TITLE
Problem running phi 3.5 in the csharp example

### ISSUE THREAD

**Issue by tranlm** [microsoft/onnxruntime-genai#855]
**Describe the bug**
Using the model builder, I've converted the phi-3.5 weights into onnx dml format. When I try to feed the resulting onnx model into the compiled binary, I get the following error:

```
C:\Users\linkm>C:\Repos\onnxruntime-genai\examples\csharp\HelloPhi\bin\Debug\net6.0\HelloPhi.exe -i -m C:\Repos\Phi-3.5-mini-instruct-int4-onnx-directml
-------------
Hello, Phi!
-------------
Model path: C:\Repos\Phi-3.5-mini-instruct-int4-onnx-directml
Interactive: True
2024-08-29 17:07:28.9214872 [E:onnxruntime:onnxruntime-genai, inference_session.cc:1842 onnxruntime::InferenceSession::Initialize] This session cannot use the graph capture feature as requested by the user as the model has control flow nodes which can't be supported by DmlExecutionProvider
Unhandled exception. Microsoft.ML.OnnxRuntimeGenAI.OnnxRuntimeGenAIException: This session cannot use the graph capture feature as requested by the user as the model has control flow nodes which can't be supported byDmlExecutionProvider
   at Microsoft.ML.OnnxRuntimeGenAI.Model..ctor(String modelPath)
   at Program.<Main>$(String[] args) in C:\Repos\onnxruntime-genai\examples\csharp\HelloPhi\Program.cs:line 52

C:\Users\linkm>C:\Repos\onnxruntime-genai\examples\csharp\HelloPhi\bin\Debug\net6.0\HelloPhi.exe -i -m C:\Repos\Meta-Llama-3.1-8B-Instruct-int4-onnx-directml
```

**To Reproduce**
Steps to reproduce the behavior
1. Go to 'examples\csharp\HelloPhi'
2. Build the solution in the folder
3. Convert the phi-3.5 model to onnx using the onnxruntime_genai.models.builder
4. Feed the directory of the resulting model to the HelloPhi.exe

**Expected behavior**
It should successfully load the model and prompt me for some input.

**Screenshots**
NA

**Desktop (please complete the following information):**
 - OS: Windows 11

**Comment by natke** [comment]
Hi @tranlm, which version of `Microsoft.ML.OnnxRuntimeGenAI` are you using?

**Comment by tranlm** [comment]
> Hi @tranlm, which version of `Microsoft.ML.OnnxRuntimeGenAI` are you using?
> 
> I cannot reproduce your error with version 0.4.0

Thanks for looking into this. I was running 0.3.0 for the `Microsoft.ML.OnnxRuntimeGenAI.DirectML` package. Just tried updating to 0.4.0 (along with 1.19.1 for `Microsoft.ML.OnnxRuntime.DirectML` and am still getting the same error. Do you think it has something to do with me using dml?

**Comment by elephantpanda** [comment]
I'm hoping they will release the Phi 3.5 version in onnx int4 format at some point. Then it'll probably work better.

**Comment by natke** [comment]
> Thanks for looking into this. I was running 0.3.0 for the `Microsoft.ML.OnnxRuntimeGenAI.DirectML` package. Just tried updating to 0.4.0 (along with 1.19.1 for `Microsoft.ML.OnnxRuntime.DirectML` and am still getting the same error. Do you think it has something to do with me using dml?

@tranlm I can run Phi 3.5 (generated with model builder), with DirectML, with the latest HelloPhi sample, using version 0.4.0.

One thing to check is that you are not running the AnyCPU target. Are you running with Visual Studio or on the command line?

Can you print out the contents of your build folder, like so:

![image](https://github.com/user-attachments/assets/e8b79447-0cea-4bca-9658-4c87fe5ef0e5)

**Comment by natke** [comment]
> I'm hoping they will release the Phi 3.5 version in onnx int4 format at some point on [Huggingface](https://huggingface.co/collections/microsoft/phi-3-6626e15e9585a200d2d761e3). Then it'll probably work better.

@elephantpanda Yes this in progress, currently going through Responsible AI and safety approvals

**Comment by tranlm** [comment]
> > Thanks for looking into this. I was running 0.3.0 for the `Microsoft.ML.OnnxRuntimeGenAI.DirectML` package. Just tried updating to 0.4.0 (along with 1.19.1 for `Microsoft.ML.OnnxRuntime.DirectML` and am still getting the same error. Do you think it has something to do with me using dml?
> 
> @tranlm I can run Phi 3.5 (generated with model builder), with DirectML, with the latest HelloPhi sample, using version 0.4.0.
> 
> One thing to check is that you are not running the AnyCPU target. Are you running with Visual Studio or on the command line?
> 
> Can you print out the contents of your build folder, like so:
> 
> ![image](https://private-user-images.githubusercontent.com/3302433/363787391-e8b79447-0cea-4bca-9658-4c87fe5ef0e5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU0NTMxMTYsIm5iZiI6MTcyNTQ1MjgxNiwicGF0aCI6Ii8zMzAyNDMzLzM2Mzc4NzM5MS1lOGI3OTQ0Ny0wY2VhLTRiY2EtOTY1OC00Yzg3ZmU1ZWYwZTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDRUMTIyNjU2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTdkMzZhM2I2NDY1ZGE2MDEyN2EyYWY3NmIzYWFmNDJlYjYxNjAxMzhhY2FmZTg5MGRjYTQxZDk4OWE1Zjk1YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.11Vy8fZ21AF6DPXIFJLC_tbabxNyvcZFVazMuuvAAQI)

Thanks @natke. I'm using Visual Studio to compile and running the exe in cmd. I noticed that you're running the Release_DirectML build (whereas before I was running the Debug version). I've switched to it and have also modified the configuration to be x64 rather than AnyCPU.  For runtime, I'm both deploying from Visual Studio and running the exe from cmd. Here's what my build folder looks like:

![image](https://github.com/user-attachments/assets/dd115071-dc4e-4090-8d91-9276700c8af2)

**Comment by natke** [comment]
Hi @tranlm, I can see that you don't have the DirectML.dlls in your build folder, which may lead to the issue you are seeing.

This is what I see in my Visual Studio solution explorer. Can you share same? 
![image](https://github.com/user-attachments/assets/6f0f88a7-92c4-4995-acf3-63738b3c18e3)

**Comment by natke** [comment]
@tranlm any updates on this?

**Comment by tranlm** [comment]
Thanks for the follow up @natke. I looked and do not have the dll's and pdb's in the project. Interestingly, the example project doesn't show any in this repo either. However, I was able to find the DirectML.Debug.dll and DirectML.dll files in my System32 folder. I went and copied them over (before and aftrer screenshots attached) to the project and tried rebuilding. But I'm still getting the same error.

Before
<img width="296" alt="Screenshot 2024-09-06 080219" src="https://github.com/user-attachments/assets/5f2de249-5cf2-41ad-9338-063738cf0e13">

After
![image](https://github.com/user-attachments/assets/2f464bbe-b353-43bb-b6d8-847e2f5f2e11)

My error
![image](https://github.com/user-attachments/assets/22a9d502-3095-416e-8e2d-ffdba0926557)
